packages/clay-core/src/tree-view/useMultipleSelection.tsx

```diff
@@ -165,6 +169,43 @@ export function useMultipleSelection<T extends Record<string, any>>(
 		}
 	}, [selectedKeys]);
 
+	/**
+	 * This `useEffect` makes `useMemo` been called every time the items change and
+	 * this can cause some performance issues. We cannot change the `useMemo` because
+	 * it's needed in other treeview cases. This needs improvements in the future.
+	 */
+
```

Comment:
LGTM... just some wordsmithing on the comment:
"This `useEffect` causes `useMemo` to be called every time the items change which can cause performance issues. We cannot change the `useMemo` because it is needed in other treeview cases. This should be improved in the future."